### PR TITLE
remove unused docs deps, bump remaining

### PR DESCRIPTION
**Pull request by pacrob**
### What was wrong?

Some docs dependencies aren't used, others can be bumped.

### How was it fixed?

Removed and bumped versions.
While here, removed deprecated use of distutils Version class.

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

image

### CODE REVIEW

**Review by fselmo — DISMISSED** [review]
I absolutely love this. Thank you 🥇 We've neglected this cleanup for too long.

**Comment by kclowes** [comment]
If you're feeling up for it, there is a deprecation warning when running `make docs` in web3/_utils/normalizers.py that I believe can just be removed. Also fine kicking it down the road too. Warning looks like: ```/Users/eve/ef/web3.py/web3/_utils/normalizers.py:242: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  if LooseVersion(eth_abi.__version__) < LooseVersion("2"):```

File: Makefile [hunk 1]
```diff
@@ -44,7 +44,6 @@ build-docs:
 	$(MAKE) -C docs doctest
 
 docs: build-docs validate-docs
-	open docs/_build/html/index.html
```
Comment (fselmo) [inline Makefile:47]: Not sure why this is an issue in your local environment but it works on my end. I don't see a need to remove this.

Comment (pacrob) [reply Makefile:47]: If you run `make docs` locally, you don't see the very last lines of output as: 
```
open docs/_build/html/index.html
make: open: Command not found
make: *** [Makefile:47: docs] Error 127
```

Comment (pacrob) [reply Makefile]: Meh, it's not there in the CI `docs` run. Un-deleted.

**Review by kclowes — APPROVED** [review]
:tada:

**Review by fselmo — APPROVED** [review]
lgtm 👍🏼

**Pull request merged**